contracts/eosio.wrap/include/eosio.wrap/eosio.wrap.hpp

```diff
@@ -6,35 +6,24 @@
 
 namespace eosio {
    /**
-    * The `eosio.wrap` system contract allows block producers to bypass authorization checks or run privileged actions with 15/21 producer approval and thus simplifies block producers superuser actions. It also makes these actions easier to audit.
-    *
+    * \details The `eosio.wrap` system contract allows block producers to bypass authorization checks or run privileged actions with 15/21 producer approval and thus simplifies block producers superuser actions. It also makes these actions easier to audit.
     * It does not give block producers any additional powers or privileges that do not already exist within the EOSIO based blockchains. As it is implemented, in an EOSIO based blockchain, 15/21 block producers can change an account's permissions or modify an account's contract code if they decided it is beneficial for the blockchain and community. However, the current method is opaque and leaves undesirable side effects on specific system accounts, and thus the `eosio.wrap `contract solves this matter by providing an easier method of executing important governance actions.
-    *
     * The only action implemented by the `eosio.wrap` system contract is the `exec` action. This action allows for execution of a transaction, which is passed to the `exec` method in the form of a packed transaction in json format via the 'trx' parameter and the `executer` account that executes the transaction. The same `executer` account will also be used to pay the RAM and CPU fees needed to execute the transaction.
     */
    class [[eosio::contract("eosio.wrap")]] wrap : public contract {
       public:
          using contract::contract;
 
          /**
-          * Execute action.
-          *
-          * @details Execute a transaction while bypassing regular authorization checks.
-          *
-          * Preconditions:
-          * - Requires authorization of eosio.wrap which needs to be a privileged account.
-          *
-          * Postconditions:
-          * - Deferred transaction RAM usage is billed to 'executer'
-          *
-          * @param executer - account executing the transaction,
-          * @param trx - the transaction to be executed.
-          *
-          * @pre
+          * \details Execute a transaction while bypassing regular authorization checks.
+          * \preconditions Requires authorization of eosio.wrap which needs to be a privileged account.
+          * \effects Deferred transaction RAM usage is billed to 'executer'.
+          * 
+          * \param executer The account executing the transaction.
+          * \param trx The transaction to be executed.
           */
-         [[eosio::action]]
-         void exec( ignore<name> executer, ignore<transaction> trx );
+         [[eosio::action]] void exec( ignore<name> executer, ignore<transaction> trx );
```

Comment:
this type of change is a must, otherwise standardese chokes with a critical error and aborts its execution, that is, the `[[eosio::action]]` has to be on the same line with the function declaration.

Reply:
@larryk85 Does this cause issues? There is no way around this requirement when using Standardese. When the C++ style declarations are multi-line Standardese does not find the entity. 

It could be possible to fix this, but not sure if we have the resources. Are multi-line declaration/entity signatures a compiler requirement?

Reply:
They are not a compiler requirement, the attributes are only lexically linked to the first binding token.  So you can place them on the same line or 100 lines between them, the compiler doesn't care.  That is odd that standardese would fail on that, it should be using Clang to parse the code and working off of the AST that it generates. I would've expected it to fail in both cases then.

Reply:
@larryk85 @dskvr 
status: if we want standardese annotations we have to either live with this constraint mentioned above (eosio action attribute has to be on the same line with the function declaration) or we should find a way in our clang patch to address this matter.
which one should we with?